src/console/buffer_printer.rs

```diff
@@ -0,0 +1,87 @@
+use std::fmt;
+use crate::console::Printer;
+use colored::*;
+
+pub struct Group {
+    collapsed: bool,
+}
+
+pub(crate) struct BufferPrinter {
+    groups: Vec<Group>,
+    buffer: String,
+}
+
+impl BufferPrinter {
+    pub fn new() -> BufferPrinter {
+        BufferPrinter {
+            groups: vec![],
+            buffer: "".to_string(),
+        }
+    }
+}
+
+impl Printer for BufferPrinter {
+    fn print(&mut self, log_level: String, args: &[&dyn fmt::Display], _options: &[&str])
```

Comment:
`log_level` should probably be an enum.  Maybe something like:

```rust
enum LogLevel {
    Debug,
    Info,
    Warn,
    Error,
}
```

Reply:
Also, keep in mind there's a [canonical logging library](https://docs.rs/log/latest/log/) that we might want to make use of as well.